Display winrate in separate box when in pondering mode.

bittsitt opened the pull request:
I added a winrate with bars that show black and whites winning chance, and more importantly, the change in this for the last move. This makes it easier to spot when you make a mistake (large negative change in win chance) and how "big" the mistake was. I considered adding color coding (orange for medium mistake, red for big mistake), but didn't add this now.

I didn't ask or make a feature request first, but I think this is useful.

![image](https://user-images.githubusercontent.com/38073282/38438054-a2f935d6-39d9-11e8-873f-e64531d5bafa.png)

dfannius commented:
While we're on the subject, one thing I have done in my local copy is to make the win rates on the board always be from Black's point of view, even when it is White's turn. This means that I don't have to keep subtracting from 100 in my head to see how it has changed from turn to turn, and matches the way that chess engines usually display their evaluation (+ is good for White, - is good for Black). I assume not everyone would like that change, but maybe we can make it configurable.

I wonder if this sort of display might look more elegant as a single vertical bar of fixed length, with the bottom 43% of it (in this case) colored black and the top 57% colored white. It should probably be an option in any case, as some people probably won't want to be distracted by it.

bittsitt commented:
One bar is a good idea. I was also thinking about a way to make a graphical representation of last move winrate change (which I think is the most important feature - was the last move a big mistake or not?) 

Maybe a red line that shows where it was before?

bittsitt commented:
Yeah, looks better with one bar. Is a line intuitive, or just confusing?

![image](https://user-images.githubusercontent.com/38073282/38447158-10939844-39fc-11e8-95b0-6e665940b97f.png)

featurecat commented:
This looks really cool. I can't wait to merge it tomorrow :). Thank you. I'll mess around with the UI until I find something that looks best to me... and then ofc let other people customize it too lol. Maybe a couple options would be nice. I want to get creative with it!

cngoodboy commented:
Looks pretty good

bittsitt commented:
Warning, if anybody has pulled this, I did a force push.. Sorry about that

bittsitt commented:
I've added the winrate graph. Could maybe have been a separate PR, but since it was highly dependent on this branch, I just added it. Personally, I find it tremendously useful. It is jumping a bit up and down when you move back and forth, I suspect it is because the first winrates we get from leelaz after a move are either bad, or it is sending us winrates of previous move. Not sure which.

It would be natural to think about adding e.g. a prescan, where you load an sgf, then do prescan where you do a (limited) analysis of each move to generate the graph. Otherwise you have to step through everything to see what happened. Possible to expand on this, and I think people who use leela zero to analyze games will find this tremendously useful.

I'm currently also not displaying any variations. I think displaying all variations will just clutter too much, but showing the variation you are currently in could be interesting.

![image](https://user-images.githubusercontent.com/38073282/38770599-b15fc486-4015-11e8-91eb-16abd4bfd9d1.png)

featurecat commented:
yay! maybe we can have a mistake highlighting

featurecat commented:
in the future

bittsitt commented:
With the graph, it is sort of obvious (large drop). :-)
